### PR TITLE
Exp comparison

### PR DESCRIPTION
**Pull request by davebulaval**
Added a method to compare the results of two experiments.

### CODE REVIEW

File: poutyne/framework/experiment.py [hunk 1]
```diff
@@ -879,3 +879,43 @@ def _test(self,
             test_stats.to_csv(self.test_log_filename.format(name=name), sep='\t', index=False)
 
         return ret
+
+    def is_better_than(self, another_experiment) -> bool:
+        """
+        Compare the results of the Experiment with another experiment. To compare, both Experiments need to be
+        logged, monitor the same metric and the same monitor mod ("min" or "max").
```
Comment (freud14) [inline poutyne/framework/experiment.py:886]:
```suggestion
        logged, monitor the same metric and the same monitor mode ("min" or "max").
```

File: poutyne/framework/experiment.py [hunk 2]
```diff
@@ -879,3 +879,43 @@ def _test(self,
...
+
+        Args:
+            another_experiment (~poutyne. Experiment): Another Poutyne experiment to compare results with.
+
+        Return:
+            A bool if either or not the Experiment is better than the Experiment to compare with.
```
Comment (freud14) [inline poutyne/framework/experiment.py:892]:
```suggestion
            Whether the Experiment is better than the Experiment to compare with.
```

File: poutyne/framework/experiment.py [hunk 3]
```diff
@@ -879,3 +879,43 @@ def _test(self,
...
+        monitored_metric = self.monitor_metric
+
+        if self.monitor_mode != another_experiment.monitor_mode:
+            raise ValueError("The monitored mode is not the same between the two experiment.")
+        monitor_mode = self.monitor_mode
+
+        checkpoint = 'best' if self.monitoring else 'last'
+        self_stats = self.load_checkpoint(checkpoint, verbose=False)
+        self_monitored_metric = self_stats.get(monitored_metric)
+        self_monitored_metric_value = self_monitored_metric.to_numpy()[0]
```
Comment (freud14) [inline poutyne/framework/experiment.py:910]: Might be a better way. Maybe something like

```python
self_monitored_metric = self_stats[monitored_metric].item()
```

Comment (davebulaval) [reply poutyne/framework/experiment.py:910]: For testability, my approach is easier. Otherwise, the mocks are painful to set properly.

Comment (davebulaval) [reply poutyne/framework/experiment.py]: switched to this approach

**Pull request merged**